### DISCUSSION TITLE
Is very slow to connect to Milvus by pymilvus

### DISCUSSION THREAD

**Discussion by w74srm** [milvus-io/milvus#36210] [Q&A and General discussion]
`from pymilvus import connections, Collection

def get_collection(host, port, collection_name):
    """创建表
    input: host milvus数据库IP string,
           port milvus数据库端口 string,
           name 表名 string,
           description 表描述 string,
           fields 列名和类型
    """
    connections.connect("default", host=host, port=port)
    collection = Collection(collection_name)
    collection.load()
    return collection`
我在执行上面的代码时会卡在`connections.connect("default", host=host, port=port)`，大概3-4分钟可以连接上，直接通过attu或者milvus_cli连接都很快，这可能是什么原因造成的呢。

**Comment by yhmo** [comment]
都是基于grpc的连接，如果这几种客户端都在同一台机器上应该没什么区别。

**Reply by w74srm** [reply]
都是在同一台机器上，但确实执行上面的代码进行连接会非常慢

**Reply by yhmo** [reply]
当你调用connections.connect("default", host=host, port=port)时，它其实没有做任何耗时的操作，它会跑到这行等待：
https://github.com/milvus-io/pymilvus/blob/baaca92afdea785537419f6600abdd1227a1b98f/pymilvus/client/grpc_handler.py#L147

你可以在这行下个断点观察，这里输入默认的timeout是10秒，也就是如果10秒之后连接不成功，就直接抛异常了。
grpc.channel_ready_future(self._channel).result(timeout=timeout)


“3-4分钟可以连接” 这个行为比较可疑，要么你看看是不是防火墙或者梯子什么的导致的。
